src/segments/nba.go

```diff
@@ -0,0 +1,425 @@
+			HomeTeamLosses: 0,
+			AwayTeamWins:   0,
+			AwayTeamLosses: 0,
+		}, nil
+	}
+
+	return nil, errors.New("no scheduled game found for team within DaysOffset days")
+}
+
+// First try to get the data from the score endpoint, if that fails, try the schedule endpoint
+// The score endpoint usually goes live within 12 hours of a game starting
+func (nba *Nba) getAvailableGameData(teamName string, httpTimeout int) (*NBAData, error) {
+	// Get the info from the score endpoint
+	data, err := nba.retrieveScoreData(teamName, httpTimeout)
+	if err == nil {
+		return data, nil
+	}
+
+	// If the score endpoint doesn't have anything get data from the schedule endpoint
+	data, err = nba.retrieveScheduleData(teamName, httpTimeout)
+	if err == nil {
+		return data, nil
+	}
+
+	return nil, err
+}
+
+// Gets the data from the cache if it exists
+func (nba *Nba) getCacheValue(key string) (*NBAData, error) {
+	if val, found := nba.env.Cache().Get(key); found {
+		var nbaData *NBAData
+		err := json.Unmarshal([]byte(val), &nbaData)
+		if err != nil {
+			return nil, err
+		}
+		return nbaData, nil
+	}
+
+	return nil, errors.New("no data in cache")
+}
+
+// Gets the data from the cache for a scheduled game if it exists
+// Checks whether the game should have started and if so, removes the cache entry
+func (nba *Nba) getCachedScheduleValue(key string) (*NBAData, error) {
+	data, err := nba.getCacheValue(key)
+	if err != nil {
+		return nil, errors.New("no data in cache")
+	}
+
+	// check if the game was previously not found and we should wait to check again
+	if data.GameStatus == NotFound {
+		return nil, errors.New("no game found for team yet")
+	}
+
+	// check if the current time is after the start time of the game
+	// if so, we need to refresh the data
+	startTime, err := time.Parse("2006-01-02T15:04:05Z", data.StartTimeUTC)
+	if err != nil {
+		return nil, err
+	}
+
+	if time.Now().UTC().After(startTime) {
+		// remove the cache entry
+		nba.env.Cache().Delete(key)
+		return nil, errors.New("game has already started")
+	}
+
+	return data, nil
+}
+
+func (nba *Nba) getResult() (*NBAData, error) {
+	teamName := nba.props.GetString(TeamName, "")
+
+	cachedScheduleKey := fmt.Sprintf("%s%s", teamName, "schedule")
+	cachedScoreKey := fmt.Sprintf("%s%s", teamName, "score")
+
+	httpTimeout := nba.props.GetInt(properties.HTTPTimeout, properties.DefaultHTTPTimeout)
+
+	// How often you want to query the API to get live score information, defaults to 2 minutes
+	cacheScoreTimeout := nba.props.GetInt(properties.CacheTimeout, 2)
+
+	// Cache the schedule information for a day so we don't call the API too often
+	cacheScheduleTimeout := nba.props.GetInt(properties.CacheTimeout, 1440)
+
+	// Cache the fact a game was not found for 30 minutes so we don't call the API too often
+	cacheNotFoundTimeout := nba.props.GetInt(properties.CacheTimeout, 30)
+
+	if cacheScheduleTimeout > 0 {
+		if data, err := nba.getCachedScheduleValue(cachedScheduleKey); err == nil {
```

Comment:
@JanDeDobbeleer - now when we check the getCachedScheduleValue, if we already have a game that was not found in the cache (so we should not check again) we will actually keep going with the rest of the code because err != nil

I have some ideas for how to fix it, but figured maybe the quickest way would be for you to resolve it. Otherwise, I'm more than happy to push some changes.

Reply:
I'll check to see what we can do about that.

Reply:
also fixed